### PR TITLE
Feature/con 495/generis events

### PR DESCRIPTION
**Pull request by bartlomiejmarszal**
Created properties related to creation and deletion of properties due to task [CON-496](https://oat-sa.atlassian.net/browse/CON-496)

### CODE REVIEW

File: core/data/event/AbstractClassPropertyEvent.php [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    /** @var core_kernel_classes_Class */
+    private $class;
+
+    public function __construct(core_kernel_classes_Class $class, array $properties)
+    {
+        $this->class = $class;
+        $this->properties = $properties;
+    }
+
+    abstract public function getName();
```
Comment (gabrielfs7) [inline core/data/event/AbstractClassPropertyEvent.php:42]: You do not need this method. It is already in the interface. Please remove it :)

File: core/data/event/ClassPropertyCreatedEvent.php [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+ * Copyright (c) 2021 (original work) Open Assessment Technologies SA;
+ */
+
+declare(strict_types=1);
+
+namespace oat\generis\model\data\event;
+
+use oat\oatbox\event\Event;
+
+class ClassPropertyCreatedEvent extends AbstractClassPropertyEvent implements Event
```
Comment (gabrielfs7) [inline core/data/event/ClassPropertyCreatedEvent.php:27]: The abstraction already implement it. Please remove the implements.

```suggestion
class ClassPropertyCreatedEvent extends AbstractClassPropertyEvent
```

File: core/data/event/ClassPropertyDeletedEvent.php [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+ * Copyright (c) 2021 (original work) Open Assessment Technologies SA;
+ */
+
+declare(strict_types=1);
+
+namespace oat\generis\model\data\event;
+
+use oat\oatbox\event\Event;
+
+class ClassPropertyDeletedEvent extends AbstractClassPropertyEvent implements Event
```
Comment (gabrielfs7) [inline core/data/event/ClassPropertyDeletedEvent.php:27]: Please remote the implements. The abstraction already has it.

```suggestion
class ClassPropertyDeletedEvent extends AbstractClassPropertyEvent
```

File: test/unit/core/data/event/ClassPropertyDeletedEventTest.php [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+use oat\generis\model\data\event\ClassPropertyDeletedEvent;
+use oat\generis\test\TestCase;
+
+class ClassPropertyDeletedEventTest extends TestCase
+{
+    public function testGetters()
+    {
+        $classMock = $this->createMock(core_kernel_classes_Class::class);
+        $subject = new ClassPropertyDeletedEvent($classMock, ['propertyUri' => 'uri']);
+        $this->assertEquals($subject->getClass(), $classMock);
```
Comment (gabrielfs7) [inline test/unit/core/data/event/ClassPropertyDeletedEventTest.php:35]: Please fix inverted assertion :) 

```suggestion
        $this->assertEquals($classMock, $subject->getClass());
```

**Review by boajer — APPROVED** [review]
- [ ] New code is covered by tests (if applicable)
- [ ] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [ ] New code is respecting code style rules
- [ ] New code is respecting best practices
- [ ] New code is not subject to concurrency issues (if applicable)
- [ ] Feature is working correctly on my local machine (if applicable)
- [ ] Acceptance criteria are respected
- [x] Pull request title and description are meaningful

**Review by shpran — APPROVED** [review]
- [x] New code is covered by tests (if applicable)
- [ ] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [x] New code is respecting code style rules
- [x] New code is respecting best practices
- [x] New code is not subject to concurrency issues (if applicable)
- [ ] Feature is working correctly on my local machine (if applicable)
- [x] Acceptance criteria are respected
- [x] Pull request title and description are meaningful

**Review by mike-ionut-mihai-sandu-tao — APPROVED** [review]
- [x] New code is covered by tests (if applicable)
- [x] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [x] New code is respecting code style rules
- [x] New code is respecting best practices
- [ ] New code is not subject to concurrency issues (if applicable)
- [x] Feature is working correctly on my local machine (if applicable)
- [x] Acceptance criteria are respected
- [ ] Pull request title and description are meaningful

**Review by gabrielfs7 — APPROVED** [review]

**Pull request merged**